packages/driver/cypress/e2e/commands/sessions/sessions.cy.js

```diff
@@ -104,248 +88,215 @@ describe('cy.session', { retries: 0 }, () => {
         })
         .url('about:blank')
       })
-    })
 
-    describe('testIsolation=legacy', { testIsolation: 'legacy' }, () => {
-      it('does not clear page', () => {
-        cy.visit('/fixtures/form.html')
-        .then(async () => {
-          cy.spy(Cypress, 'action').log(false)
+      it('clears session data before each run', async () => {
+        const clearCurrentSessionData = cy.spy(Cypress.session, 'clearCurrentSessionData')
 
-          await Cypress.action('runner:test:before:run:async', {})
+        await Cypress.action('runner:test:before:run:async', {})
 
-          expect(Cypress.action).not.to.be.calledWith('cy:url:changed')
-          expect(Cypress.action).not.to.be.calledWith('cy:visit:blank')
-        })
-        .url('/fixtures/form.html')
+        expect(clearCurrentSessionData).to.be.called
       })
-    })
-  })
 
-  describe('session flows', () => {
-    let logs = []
-    let clearPageCount = 0
-    let sessionGroupId
-    let setup
-    let validate
+      it('resets rendered html origins before each run', async () => {
+        const backendSpy = cy.spy(Cypress, 'backend')
 
-    const handleSetup = () => {
-      // create session clears page before running
-      cy.contains('Default blank page')
-      cy.contains('This page was cleared by navigating to about:blank.')
+        await Cypress.action('runner:test:before:run:async', {})
 
-      cy.visit('/fixtures/auth/index.html')
-      cy.contains('You are not logged in')
-      cy.window().then((win) => {
-        win.sessionStorage.setItem('cypressAuthToken', JSON.stringify({ body: { username: 'tester' } }))
+        expect(backendSpy).to.be.calledWith('reset:rendered:html:origins')
       })
-    }
 
-    const handleValidate = () => {
-      // both create & restore session clears page after running
-      cy.contains('Default blank page')
-      cy.contains('This page was cleared by navigating to about:blank.')
+      it('clears the browser context before each run', () => {
+        cy.window()
+        .then((win) => {
+          win.cookie = 'key=value; SameSite=Strict; Secure; Path=/fixtures'
+          win.localStorage.setItem('animal', 'bear')
+          win.sessionStorage.setItem('food', 'burgers')
```

Comment:
I wonder if we need a similar test check if `document.cookie` and the cookie-jar are cleared between tests when going back to `cy.origin`?